File: 3term/1.2. local network/1.2. local network/StuffClass.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _1._2.local_network
{
    /// <summary>
    /// Class which provide interaction of system with input data.
    /// </summary>
    public class StuffClass
    {
        private StreamReader reader = new StreamReader("info.txt");
        private Computer[] computers;
        private int[,] stuffMatrix;
        private int size = 4;

        /// <summary>
        /// Start work.
        /// </summary>
        public StuffClass()
        {
            this.Start();
        }

        /// <summary>
        /// Start of infection process.
        /// </summary>
        public void Start()
        {
            string counter = reader.ReadLine();
            size = Convert.ToInt32(counter);

            computers = new Computer[size];

            bool isInfected = false;
            string name = "";
            string systemType = "";

            for (int i = 0; i < size; ++i)
            {
                computers[i] = new Computer();
                string sign = reader.ReadLine();
                if (sign == "+")
                {
                    isInfected = true;
                }
                else if (sign == "-") 
                {
                    isInfected = false;
                }
                name = reader.ReadLine();
                systemType = reader.ReadLine();

                computers[i].SetHealthCondition(isInfected); ;
                computers[i].SetOSType(computers[i].InfectionIndex(systemType));
                computers[i].SetName(name);
            }

            stuffMatrix = new int[size, size];
            for (int i = 0; i < size; ++i)
            {
                for (int j = 0; j < size; ++j)
                {
                    stuffMatrix[i, j] = 0;
                }
            }

            for (int i = 0; i < size; ++i)
            {
                string str = reader.ReadLine();
                for (int j = 0; j < size; ++j)
                {
                    stuffMatrix[i, j] = (int)Char.GetNumericValue(str[j]);
                }
            }

        }

        /// <summary>
        /// Get computers's array.
        /// </summary>
        /// <returns></returns>
        public Computer[] GetArrayOfComputers()
        {
            return computers;
        }

        /// <summary>
        /// Get matrix of system interaction.
        /// </summary>
        /// <returns></returns>
        public int[,] GetMatrix()
        {
            return stuffMatrix;
        }
    }
}

File: 4term/5.1. localNetwork/5.1. localNetwork/StuffClass.fs
module StuffClass
open System
open System.Collections.Generic
open System.Linq
open System.Text
open System.Threading.Tasks
open System.IO
open Microsoft.FSharp.Collections
open Computer

/// Class which provide interaction of system with input data.
type StuffClass (info : (string * string * string) array, matrixArr : int [] []) = 
     let mutable compList : Computer list = []
     let mutable size : int = 5
     let mutable stuffMatrix = Array2D.create size size 0


/// Start of infection process.
     member s.Start() = 
         //size <- int info.Head
         size <- info.Length
 
         //for i in 1 .. 3 .. (size * 3) do
         for i = 0 to 2 do
             let comp = new Computer()
             //let sign : string = info.[i]
             let (fst, snd, trd) = info.[i]
             let sign : string = fst
             let isInfected = 
                 sign = "+"
             let name = snd
             let systemType = trd
              
             comp.SetHealthCondition(isInfected)
             comp.SetOSType(comp.InfectionIndex(systemType))
             comp.SetName(name)
              
             let someList = compList @ [comp]
             compList <- someList
              
         stuffMatrix <- Array2D.create size size 0
         for i = 0 to (size - 1) do
             for j = 0 to (size - 1) do
                 stuffMatrix.[i, j] <- matrixArr.[i].[j]

/// Get computers's list.
     member s.GetArrayOfComputers() = compList


/// Get matrix of system interaction.
     member s.GetMatrix() = stuffMatrix


        
